instrumentation/lettuce/lettuce-5.0/src/main/java/io/opentelemetry/auto/instrumentation/lettuce/v5_0/ConnectionFutureAdvice.java

```diff
@@ -16,41 +16,39 @@
 
 package io.opentelemetry.auto.instrumentation.lettuce.v5_0;
 
-import static io.opentelemetry.trace.Span.Kind.CLIENT;
-import static io.opentelemetry.trace.TracingContextUtils.currentContextWith;
+import static io.opentelemetry.auto.instrumentation.lettuce.v5_0.LettuceDatabaseClientTracer.TRACER;
 
 import io.lettuce.core.ConnectionFuture;
 import io.lettuce.core.RedisURI;
-import io.opentelemetry.auto.instrumentation.api.SpanWithScope;
+import io.opentelemetry.context.Scope;
 import io.opentelemetry.trace.Span;
 import net.bytebuddy.asm.Advice;
 
 public class ConnectionFutureAdvice {
 
   @Advice.OnMethodEnter(suppress = Throwable.class)
-  public static SpanWithScope onEnter(@Advice.Argument(1) final RedisURI redisURI) {
-    final Span span =
-        LettuceClientDecorator.TRACER.spanBuilder("CONNECT").setSpanKind(CLIENT).startSpan();
-    LettuceClientDecorator.DECORATE.afterStart(span);
-    LettuceClientDecorator.DECORATE.onConnection(span, redisURI);
-    return new SpanWithScope(span, currentContextWith(span));
+  public static void onEnter(
+      @Advice.Argument(1) final RedisURI redisURI,
+      @Advice.Local("otelSpan") Span span,
+      @Advice.Local("otelScope") Scope scope) {
+    span = TRACER.startSpan("CONNECT", redisURI, null);
+    scope = TRACER.startScope(span);
   }
 
   @Advice.OnMethodExit(onThrowable = Throwable.class, suppress = Throwable.class)
   public static void stopSpan(
-      @Advice.Enter final SpanWithScope spanWithScope,
       @Advice.Thrown final Throwable throwable,
-      @Advice.Return final ConnectionFuture<?> connectionFuture) {
-    final Span span = spanWithScope.getSpan();
+      @Advice.Return final ConnectionFuture<?> connectionFuture,
+      @Advice.Local("otelSpan") Span span,
+      @Advice.Local("otelScope") Scope scope) {
+    if (scope != null) {
+      scope.close();
+    }
```

Comment:
conditional not needed(?)
```suggestion
    scope.close();
```

Reply:
I used the same pattern as we have in `HttpServerTracer`. And the same pattern will be used later in `HttpClientTracer`. But I can remove it here.

Reply:
i think remove, otherwise the reader wonders the question when can scope be `null`